rbd/clone_image_by_id.go

```diff
@@ -30,17 +49,26 @@ func CloneImageByID(ioctx *rados.IOContext, parentName string, snapID uint64,
 		return rbdError(C.EINVAL)
 	}
 
+	rbdClone4, err := dlsym.LookupSymbol("rbd_clone4")
```

Comment:
If we really want to go this route, and even more such cases are coming, I would really prefer to either
- have init() code, that initializes the function pointer once at startup time, or
- use once.Do() to initialize it one time at first call.

Reply:
What about making `rbdClone4` a global variable, if `nil` lookup the symbol, use it otherwise?

Reply:
Updated this PR to use `sync.Once` for checking with `dlsym` for `rbd_clone4`. Tested against Ceph **main** (PASS) and Ceph **pacific** (SKIPPED, expected).